### PR TITLE
First implementation of smime calls

### PR DESCRIPTION
**Pull request by joachimBurket**
- Implemented: list, list_by_email, enroll, collect, replace, revoke, revoke_by_email
- Modified the version_hack() function in the helpers in order to use it

### CODE REVIEW

**Comment by joachimBurket** [comment]
I am using your library to manage SMIME certificates on Sectigo. I have started to code the SMIME API calls, and use it for a first dev version.

I had to do some modifications on the version_hack() function in the helpers in order to use it.

I also think some updates on the `Certificate` base class might be needed in order to reuse code.

Finally, I wasn't sure about how the Exceptions are handled. The Sectigo API is returning the detail of the error into the body in the form
```json
{"code":-188,"description":"You already have allowed number of valid certificates."}
```
Should the error handling be left to the user of the library, recovering the error detail in the HTTPError exception, or should the error handling be done into the lib, and a custom exception raised depending on the parsed error.

**Comment by coreone** [comment]
Looks good to me so far, and thanks for that change to `version_hack`.  I think I wrote that for a function that is no longer in the repo when the structure was a bit different.

For exceptions, I was originally just using requests' `raise_for_status` simply to raise exceptions if non 200-level statuses were returned. I haven't had a chance to address it yet, but #28 was kind of getting at the same issue. I would tend to just use the `HTTPError.response.json()` to pull out that information if needed.

**Comment by coreone** [comment]
If possible, adding unit tests for this new code would be much appreciated.

**Comment by joachimBurket** [comment]
Okay thanks for the info for the Exceptions. I'll keep it how it is.

For the tests I wanted to wait for your feedback before starting them. I'll make an update those next days.

**Comment by joachimBurket** [comment]
I've added the unit tests for the SMIME code. Tell me if something is missing

**Comment by coreone** [comment]
This looks good, with a few bugs commented inline.  Also, there are a bunch of lint/code style errors that appear, which you can see at:

https://github.com/broadinstitute/python-cert_manager/pull/39/checks?check_run_id=3221026125

**Comment by joachimBurket** [comment]
I ran pylama in local, I think the linting errors are good now. I had to disable R0914 (Too many local variables) because of the enroll() method (had 22 instead of 15 local vars). Are you okay with that or does it need a refactor ?

**Comment by coreone** [comment]
Looks like there are still a bunch that are mostly whitespace-related under the `tests` directory.  Only other lint error seems to be in `test_smime.py` on line 143, but you should just be able to add `# pylint: disable=too-many-instance-attributes` right above `class TestEnroll(TestSMIME):` to ignore it.

**Comment by joachimBurket** [comment]
Sorry I forgot to run pylama on the test folder.. Hope its fine now ;)

**Review by coreone — CHANGES_REQUESTED** [review]
@joachimBurket Looks like the linting is clear now, but there are still the two places I commented on `Pending` being used instead of `Revoked`.

**Comment by coreone** [comment]
Additionally, there have been some minor merges, so if you could rebase as well.

File: tests/test_smime.py [hunk 1]
```diff
@@ -53,3 +57,456 @@ def test_version(self):
...
+    @responses.activate
+    def test_revoked(self):
+        """It should raise a Revoked exception if an Http error with the pending code in the body is returned."""
+        # Setup the mocked response
+        body = json.dumps({"code":Revoked.CODE,"description":"The Certificate has been revoked!"})
+        responses.add(responses.GET, self.test_url, body=body, status=400)
+
+        # Call the function, expecting an exception
+        smime = SMIME(client=self.client)
+        self.assertRaises(Pending, smime.collect, self.test_id)
```
[the remark this answers is not in the document]
Comment (joachimBurket) [reply tests/test_smime.py:348]: Ouch didn't see that :O yes it should, I update it

**Review by coreone — APPROVED** [review]

**Comment by coreone** [comment]
Looks good.  Merging in.  Thanks for the contribution!!

**Comment by joachimBurket** [comment]
You're welcome! :)

**Pull request merged**